volumes/vds/py/vds_functions.py

```diff
@@ -0,0 +1,464 @@
+import logging
+import pandas as pd
+from numpy import nan
+from psycopg2 import sql, Error
+from psycopg2.extras import execute_values
+import struct
+from datetime import datetime
+import pytz
+from airflow.macros import ds_add
+from airflow.models import Variable
+from airflow.hooks.base import BaseHook
+from airflow.providers.slack.operators.slack_webhook import SlackWebhookOperator
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+# connection to slack
+SLACK_CONN_ID = 'slack_data_pipeline'
+
+def task_fail_slack_alert(dag_name, owners, context):
+    # connection to slack
+    global SLACK_CONN_ID
+    
+    slack_ids = Variable.get('slack_member_id', deserialize_json=True)
+    list_names = []
+    for name in names:
+        list_names.append(slack_ids.get(name, '@Unknown Slack ID')) #find slack ids w/default = Unkown
+
+    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
+    
+    log_url = context.get('task_instance').log_url.replace(
+        'localhost', context.get('task_instance').hostname + ":8080"
+    )
+    
+    slack_msg = """
+        :ring_buoy: {dag}.{task} Task Failed.         
+        *Log Url*: {log_url}
+        {slack_name} please check.
+        """.format(
+            task=context.get('task_instance').task_id,
+            dag=context.get('task_instance').dag_id,
+            exec_date=context.get('execution_date'),
+            log_url=log_url,
+            slack_name=' '.join(list_names)
+    )
+    
+    failed_alert = SlackWebhookOperator(
+        task_id='slack_test',
+        http_conn_id='slack',
+        webhook_token=slack_webhook_token,
+        message=slack_msg,
+        username='airflow',
+        proxy='http://'+BaseHook.get_connection('slack').password+'@137.15.73.132:8080',
```

Comment:
I updated the VFH code and Airflow `slack` connection to avoid hard-coding the proxy IP address. Check it [here](https://github.com/Toronto-Big-Data-Innovation-Team/vfh_data_mgmt/blob/133d989ba882e0d3844aa98988b15e7ab5757663/utils/dag_functions.py#L171-L174) if you want to apply it to this function